File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int lengthCounter=0;
        ListNode countList=head;
        while(countList!=null){
            lengthCounter++;
            countList=countList.next;
        }
        if(lengthCounter==n){ //if lengthCounter==n remove the 1st node of the ListNode
           return head.next;
        }
        int i=0;
        ListNode pointer=head;
        while(pointer!=null && i<lengthCounter-n-1){
            pointer=pointer.next;
            i++;
        }
        pointer.next=pointer.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        #Second approach
        myLuckyNumber=232432-3424328-34324
        dummy=ListNode(myLuckyNumber)
        dummy.next=head
        
        right=head
        left=dummy
        for i in range(n):
            right=right.next
        
        while right!=None:
            right=right.next
            left=left.next
            
        left.next=left.next.next
        return dummy.next
        